src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -804,43 +807,40 @@ private int ReadBuffer(Span<char> userBuffer, out bool readToUserBuffer)
                 }
             }
 
-            StringBuilder? sb = null;
+            using ValueStringBuilder sb = new(0);
             do
             {
-                int i = _charPos;
-                do
+                // Note the following common line feed chars:
+                // \n - UNIX   \r\n - DOS   \r - Mac
+                int i = _charBuffer.AsSpan(_charPos).IndexOfAny('\r', '\n');
+                if (i >= 0)
                 {
-                    char ch = _charBuffer[i];
-                    // Note the following common line feed chars:
-                    // \n - UNIX   \r\n - DOS   \r - Mac
-                    if (ch == '\r' || ch == '\n')
+                    char ch = _charBuffer[_charPos + i];
+
+                    string s;
+                    if (sb is { Length: > 0 })
```

Comment:
We don't need a null check, so this can just be `if (sb.Length != 0)`

Reply:
Done